controllers/controllers.js

```diff
@@ -21,4 +21,26 @@ exports.getArticleById=(req,res,next) => {
         return res.status(200).send({article});
     });
 
-};
\ No newline at end of file
+};
+
+exports.patchArticleById=(req,res,next) =>{
+    
+    const {article_id} =req.params;
+    const {inc_votes} = req.query;
+
+    if(Number.isNaN(+article_id)){
+        next({status:400, msg:'Invalid article_id'})
+        return ;
+    }
+
+    //Using && short-circuit to avoid .length of undefined
+    if(typeof req.query==='object' && Array.isArray(req.query)===false &&
+    Object.keys(req.query).length===1 && Number.isInteger(+inc_votes)){
+        return models.updateArticleById(inc_votes,article_id).then(({rows:[article]})=>{
+            return res.status(200).send({article});
+        }).catch((err) => next(err))
+    } else {
+        next({status:400, msg:'Invalid PATCH body'})
```

Comment:
Same as previous comment, `Promise.reject` here and keep the `next` in the catch block.